engine/src/main/junit-shared/org/operaton/bpm/engine/test/util/JobExecutorHelper.java

```diff
@@ -0,0 +1,320 @@
+   * @param maxMillisToWait   the maximum time in milliseconds to wait for all jobs to be processed
+   * @param checkInterval     the interval in milliseconds to check for job completion
+   * @param jobExecutor       the {@link JobExecutor} responsible for processing the jobs
+   * @param managementService the {@link ManagementService} used to interact with jobs management
+   * @param shutdown          a flag indicating whether to shut down the job executor after processing
+   */
+  public static void waitForJobExecutorToProcessAllJobs(String processInstanceId,
+                                                        long maxMillisToWait,
+                                                        long checkInterval,
+                                                        JobExecutor jobExecutor,
+                                                        ManagementService managementService,
+                                                        boolean shutdown) {
+    activateJobExecutor(jobExecutor);
+    maxMillisToWait = calculateMaxWaitTime(maxMillisToWait, jobExecutor);
+    try {
+      Callable<Boolean> condition = () -> !areJobsAvailableByPiId(processInstanceId, managementService);
+      waitForCondition(condition, maxMillisToWait, checkInterval);
+    } catch (Exception e) {
+      throw new ProcessEngineException("Time limit of " + maxMillisToWait + " was exceeded.", e);
+    } finally {
+      if (shutdown) {
+        jobExecutor.shutdown();
+      }
+    }
+  }
+
+  /**
+   * Activates the specified job executor by starting it if it is not already active.
+   * This method checks the current state of the job executor and ensures that it
+   * transitions to an active state if it is inactive.
+   *
+   * @param jobExecutor the {@link JobExecutor} instance to be activated
+   */
+  private static void activateJobExecutor(JobExecutor jobExecutor) {
+    // Ensure the job executor is inactive before proceeding
+    if (jobExecutor.isActive()) {
+      return;
+    }
+    jobExecutor.start();
+  }
+
+  /**
+   * Determines if there are jobs available for execution associated with a specific process instance ID.
+   * The method checks for executable, active jobs with retries left through the provided ManagementService.
+   *
+   * @param processInstanceId the ID of the process instance for which jobs should be checked
+   * @param managementService the ManagementService used to query and manage jobs
+   * @return true if jobs are available for the specified process instance, false otherwise
+   */
+  private static boolean areJobsAvailableByPiId(String processInstanceId, ManagementService managementService) {
+    return managementService.createJobQuery()
+        .withRetriesLeft()
+        .executable()
+        .active()
+        .processInstanceId(processInstanceId)
+        .count() > 0;
+
+  }
+
+  /**
+   * Checks if there are jobs available for processing using the provided ManagementService.
```

Comment:
```suggestion
   * Checks if there are jobs available for processing using the provided {@link ManagementService}.
```

Reply:
Please also look further for usages of specific class names in the docs and use links